governance/delegation-interop.md

```diff
@@ -0,0 +1,65 @@
+number of the latest checkpoint. The contract on OP mainnet will then consume these events, and in its own storage provide a mapping between chainIds and an object containing both
+the latest checkpoint and the block number. After tokens are transferred using the `GovToken` contract a check is performed on the delegation contract that ensures that before this
+mapping is updated, the nonce of the token transfer received has been incremented as well as being sent from a block number greater than the current one.
+
+# Problem Statement + Context
+
+In the existing `GovernanceDelegation` and `GovToken` contract it does not support interoperability, thus making the voting power fragmented across all of the potential L2 chains. The negative consequence of this is that tracking delegations becomes a much harder task when users have partially delegated to many different users that may or may not be on the same chain.
+
+However, pushing all of the logic onto the `GovToken` proves to be problematic given the potential flood of messages that may be out of sequence or have many smaller transfer events
+prior to a single delegation event. The following section aims to outline a solution to correct this as well as ensure that relayers still have the proper incentives to handle these
+messages.
+
+# Alternatives Considered
+
+When originally considering this problem, the idea was to create a hook during the `afterTokenTransfer` event. As alluded to above, this do not provide some of the desired functionality.It also makes a few assumptions about the behaviors of delegates that receive tokens and stifles their ability to create partial delegations in a low-cost manner. Moreover, if there was
+a want to later batch updates of voting power to mainnet it would be costly to those operating governance.
+
+Another active consideration is the use of block timestamp of each L2 instead of block number while implementing this new solution. The primary reason for block numbers to be used in
+its stead, stems from a concern of manipulation by sequencers and the possible delays that might need to be enforced between events to ensure correctness.
+
+# Proposed Solution
+
+## Rationale
+
+Due to the fact that the `GovernanceDelegation` contract now has awareness of the block numbers and chain IDs of all of the voting power checkpoints each L2 can now be responsible for
+their own checkpointing and thus make as many updates necessary without cost to relayers. From an implementation standpoint many of the changes are quite small and should minimize the
+risk of complication of an already intricate interface.
+
+The main goal is to ensure that the incentives are set up correctly so that relayers and indexers can manage and provide users the latest delegation changes when exercising their voting
+power.
+
+## Implementation
```

Comment:
> It'd be great to have a figma or detailed flows on how you are envisioning the technical implementation being.

+1 to this comment by @skeletor-spaceman, this would be a big help to understanding the proposal. What happens during some specific user stories (same chain token transfer, cross-chain token transfer, same-chain delegation, cross-chain delegation, voting, etc) would be really great